engine/src/font/atlas.cpp

```diff
@@ -0,0 +1,57 @@
+#include <vector>
+
+#include <msdf-atlas-gen/msdf-atlas-gen.h>
+
+#include <cubos/core/ecs/reflection.hpp>
+#include <cubos/core/gl/render_device.hpp>
+#include <cubos/core/reflection/reflect.hpp>
+#include <cubos/core/reflection/type.hpp>
+
+#include <cubos/engine/font/atlas.hpp>
+
+CUBOS_REFLECT_IMPL(cubos::engine::FontAtlas)
+{
+    return cubos::core::reflection::Type::create("cubos::engine::FontAtlas");
+}
+
+cubos::engine::FontAtlas::FontAtlas(msdfgen::FontHandle* font, core::gl::RenderDevice& rd)
+{
+    std::vector<msdf_atlas::GlyphGeometry> glyphs;
+    msdf_atlas::FontGeometry fontGeometry(&glyphs);
+    fontGeometry.loadCharset(font, 1.0, msdf_atlas::Charset::ASCII);
+
+    const double maxCornerAngle = 3.0;
+    for (msdf_atlas::GlyphGeometry& glyph : glyphs)
+    {
+        glyph.edgeColoring(&msdfgen::edgeColoringInkTrap, maxCornerAngle, 0);
+    }
+    msdf_atlas::TightAtlasPacker packer;
+    packer.setDimensionsConstraint(msdf_atlas::DimensionsConstraint::SQUARE);
+    packer.setMinimumScale(24.0);
+    packer.setPixelRange(4.0);
+    packer.setMiterLimit(1.0);
+    packer.pack(glyphs.data(), static_cast<int>(glyphs.size()));
+    int width = 0;
+    int height = 0;
+    packer.getDimensions(width, height);
+
+    msdf_atlas::ImmediateAtlasGenerator<float, 3, &msdf_atlas::msdfGenerator,
+                                        msdf_atlas::BitmapAtlasStorage<msdf_atlas::byte, 3>>
+        generator(width, height);
+    msdf_atlas::GeneratorAttributes attributes;
+    generator.setAttributes(attributes);
+    generator.setThreadCount(4);
```

Comment:
Lots of magic numbers here! Some of those should probably be in constants, and others should be configurable by the user when creating a font atlas. If we do go through that 'FontAtlasBridge' I mentioned in the other comment, I guess we could fetch those options from the .meta